osu.Desktop/Updater/AppImageUpdateManager.cs

```diff
@@ -0,0 +1,551 @@
+                }
+
+                return state;
+            }
+            private set => state = value;
+        }
+
+        /// <summary>
+        /// Absolute path to AppImage file (with symlinks resolved)<para />
+        /// See: https://docs.appimage.org/packaging-guide/environment-variables.html<para />
+        /// TODO: verify whether environment variable APPIMAGE is set in the deployed build<para />
+        /// Will Assume osu.AppImage in the <see cref="RuntimeInfo.StartupDirectory" /> otherwise
+        /// </summary>
+        public static string AppImagePath => IsAppImage ? Environment.GetEnvironmentVariable("APPIMAGE") : $"{RuntimeInfo.StartupDirectory}osu.AppImage";
+
+        /// <summary>
+        /// Checks if osu was launched as an AppImage
+        /// </summary>
+        public static bool IsAppImage => Environment.GetEnvironmentVariable("APPIMAGE") != null;
+
+        private static bool hasUpdates()
+        {
+            try
+            {
+                using (var process = new Process())
+                {
+                    process.StartInfo = startInfo();
+                    process.Start();
+
+                    process.WaitForExit();
+                    return process.ExitCode == 1;
+                }
+            }
+            catch
+            {
+                return false;
+            }
+        }
+
+        /// <summary>
+        /// Checks if there is a new release of the AppImage with the embedded update information
+        /// </summary>
+        public static bool HasUpdates => State == States.UPDATESAVAILABLE;
+
+        private static string getOutput(ToolArguments arg)
+        {
+            try
+            {
+                using (var process = new Process())
+                {
+                    process.StartInfo = startInfo(arg);
+                    process.Start();
+
+                    string output = process.StandardError.ReadToEnd();
+                    process.WaitForExit();
+                    return output;
+                }
+            }
+            catch
+            {
+                return null;
+            }
+        }
+
+        private static string version;
+
+        /// <summary>
+        /// Gets version information of the appimageupdatetool binary
+        /// </summary>
+        public static string Version
+        {
+            get
+            {
+                if (version is default(string))
+                {
+                    string output = getOutput(ToolArguments.ToolVersion);
+
+                    if (output != null)
+                    {
+                        var match = Regex.Match(output, @"version\s([a-zA-Z0-9\._-]*).*$", RegexOptions.IgnoreCase);
+
+                        if (match.Success)
+                        {
+                            output = match.Groups[1].ToString();
+                        }
+                    }
+
+                    version = output;
+                }
```

Comment:
code block needs to have less nesting than this `if { ... if { ... if {` structure. invert the checks and early-return instead to have it flatter